### ISSUE TITLE
Fix false positives in string-no-newline

### ISSUE THREAD

**Issue by FelschR** [stylelint/stylelint#3226]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug when used with HTML files.

> Which rule, if any, is this issue related to?

string-no-newline

> What file is needed to reproduce this issue?

```html
<p>this is fine.</p>
<p>this isn't.</p>
<span>this line is still affected.</span>
<div style="ttt: auto">actual issues are still reported</div>
```

> What stylelint configuration is needed to reproduce this issue?

I'm pretty sure that it's irrelevant but this is my config:
```json
{
    "extends": "stylelint-config-recommended",
    "rules": {
        "font-family-no-missing-generic-family-keyword": null
    }
}
```

> Which version of stylelint are you using?

9.1.3

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint [filename].html`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

It uses standard HTML.

> What did you expect to happen?

There should be no `string-no-newline` errors starting from line 2.

> What actually happened (e.g. what warnings or errors you are getting)?

At the end linebreak after the line with an apostrophe I get an `string-no-newline` stylelint error.
Also all following line breaks are affected.

**Comment by gucong3000** [comment]
fixed here: #3228

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/issues/3219
